### PR TITLE
Fix sdl versioning for vehicle data

### LINKED ISSUE #2997: SDL does not apply the versioning for vehicle data
### Bug Report
SDL does not apply the versioning for vehicle data

##### Preconditions
1. SDL and HMI are started
2. App has permissions for fuelRange, parameter is available `since` 5.0 version

##### Reproduction Steps
1. Register application with `majorVersion = 3, minorVersion = 0`
2. App requests `SubscribeVehicleData` with `fuelRange`

##### Expected Behavior
SDL procesess  the request with `fuelRange` as request with undefined data

##### Observed Behavior
SDL processed  the request with `fuelRange` as defined data

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/commit/8db0c2601921fdd368bf86133780746b532c40f5
* ATF: https://github.com/smartdevicelink/sdl_atf/commit/d6c2138f8d894d1e343d971226a16c47019ee92e
* Script: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2256

##### Logs
[report.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3535582/report.tar.gz)

### PR DESCRIPTION
**Pull request by YarikMamykin**
Fixes #2997 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Added unit tests for RPCHandlerImpl class (GetMessageVersion method)
Adjust ATF scripts for GetVehicleData

### Summary
Corrected processing of SyncMsgVersion according to mobile application version (on RegisterAppInterface).
Corrected validating of incoming parameters from mobile request.

### Tasks Remaining:
- [ ] [Adjust ATF scripts for GetVehicleData]

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 1]
```diff
@@ -267,11 +267,14 @@ TEST_F(RCGetInteriorVehicleDataConsentTest,
        Run_MobileSendButtonPressMessage_HMISendINUSEModeToMobile) {
   // Arrange
   auto mobile_message = CreateBasicMessage();
+  auto msg_ver = utils::SemanticVersion();
 
   // Expectations
   EXPECT_CALL(mock_allocation_manager_, AcquireResource(_, _, _))
       .WillOnce(Return(rc_rpc_plugin::AcquireResult::IN_USE));
 
+  EXPECT_CALL(*mock_app_, msg_version()).WillOnce(ReturnRef(msg_ver));
```
Comment (alexkutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @YarikMamykin if we are not testing SemanticVersion here, please use OnCall

File: src/components/smart_objects/src/object_schema_item.cc [hunk 2]
```diff
@@ -251,25 +252,23 @@ CObjectSchemaItem::CObjectSchemaItem(const Members& members)
...
+       ++it) {
     const std::string& key = it->first;
     std::map<std::string, SMember>::const_iterator members_it =
         mMembers.find(key);
-    if (mMembers.end() == members_it
-        // FIXME(EZamakhov): Remove illegal usage of filed in AM
-        && key.compare(connection_key) != 0 && key.compare(binary_data) != 0 &&
-        key.compare(app_id) != 0) {
-      ++it;
+    // FIXME(EZamakhov): Remove illegal usage of filed in AM
```
Comment (alexkutsan) [inline src/components/smart_objects/src/object_schema_item.cc]: Remove this fixme

Comment (YarikMamykin) [reply src/components/smart_objects/src/object_schema_item.cc]: @alexkutsan , fixed in 9e3c0d23c33cd5ff17ec5796ebd57e1af7a850b4

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc (hunk 1 above)
Comment (YarikMamykin) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @alexkutsan , fixed in 504844c7e81194da100c5a113c4f016acfb34b50

File: src/components/application_manager/src/rpc_handler_impl.cc [hunk 3]
```diff
@@ -296,9 +296,9 @@ void RPCHandlerImpl::GetMessageVersion(
     }
     utils::SemanticVersion temp_version(major, minor, patch);
     if (temp_version.isValid()) {
-      message_version = (temp_version >= utils::rpc_version_5)
-                            ? temp_version
-                            : utils::base_rpc_version;
+      message_version = temp_version;
+    } else {
+      message_version = utils::base_rpc_version;
```
Comment (AByzhynar) [inline src/components/application_manager/src/rpc_handler_impl.cc]: @YarikMamykin Remove redundant `else`. You can do it like here: 
```c++
message_version = utils::base_rpc_version;
utils::SemanticVersion temp_version(major, minor, patch);
if (temp_version.isValid()) {
   message_version = temp_version;
}
```

File: src/components/application_manager/test/rpc_handler_impl_test.cc [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#include <memory>
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/smart_object_keys.h"
+#include "interfaces/HMI_API.h"
+#include "interfaces/MOBILE_API.h"
+
+namespace test {
+namespace components {
+namespace application_manager_test {
+using ::test::components::application_manager_test::MockApplicationManager;
```
Comment (AByzhynar) [inline src/components/application_manager/test/rpc_handler_impl_test.cc]: @YarikMamykin 
Please add an empty line here

File: src/components/application_manager/test/rpc_handler_impl_test.cc [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      utils::SemanticVersion("2.5.0"),
+      utils::SemanticVersion("3.0.0"),
+      utils::SemanticVersion("5.0.0"),
+      utils::SemanticVersion("6.1.0")};
+
+  SmartObject message;
+  message[json_str::S_MSG_PARAMS] = SmartObject(SmartType_Map);
+  message[json_str::S_MSG_PARAMS][app_str::sync_msg_version] =
+      SmartObject(SmartType_Map);
+  utils::SemanticVersion result_message_version;
```
Comment (AByzhynar) [inline src/components/application_manager/test/rpc_handler_impl_test.cc]: @YarikMamykin 
Move it to the place before its first usage

Comment (YarikMamykin) [reply src/components/application_manager/test/rpc_handler_impl_test.cc]: @AByzhynar , fixed in 8ac65232bab1f6c381e6c8862203df4707626522

File: src/components/application_manager/test/rpc_handler_impl_test.cc (hunk 4 above)
Comment (YarikMamykin) [reply src/components/application_manager/test/rpc_handler_impl_test.cc]: @AByzhynar , fixed in 8ac6523

File: src/components/application_manager/src/rpc_handler_impl.cc (hunk 3 above)
Comment (YarikMamykin) [reply src/components/application_manager/src/rpc_handler_impl.cc]: @AByzhynar , fixed in 75c4639d3ebb0d905b7414f87efef58b50cf99d7

**Comment by YarikMamykin** [comment]
@theresalech , Luxoft internal code review is completed. Pull request is ready for the Livio team review.

File: src/components/smart_objects/src/object_schema_item.cc [hunk 6]
```diff
@@ -289,6 +288,7 @@ const SMember& CObjectSchemaItem::GetCorrectMember(
     }
   }
   // Return member as default
+  member.mIsValid = false;
```
[the remark this answers is not in the document]
Comment (YarikMamykin) [reply src/components/smart_objects/src/object_schema_item.cc]: @iCollin , please take a look at 5664c624277f299959c8596a0b51a84d911b7325

**Comment by iCollin** [comment]
Reverted by #3028 

If a mobile application reports an RPC version less than 5, it MUST be assigned `base_rpc_version`

**Pull request merged**